src/reconcile/rec_write.c

```diff
@@ -555,6 +560,14 @@ __rec_write_status(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 		return (EBUSY);
 
 	/*
+	 * If there is only one block and no update was used, no progress has
+	 * been made and there is no point swapping the new page into place.
+	 */
+	if (r->bnd_next == 1 && F_ISSET(r, WT_EVICT_UPDATE_RESTORE) &&
+	    r->bnd[0].supd != NULL && !r->update_used)
+		return (EBUSY);
+
+	/*
```

Comment:
@michaelcahill, if you make a change here, I think the code in `txn_ckpt.c` to re-mark the tree dirty could use `__wt_tree_modify_set()` instead of inlining it.